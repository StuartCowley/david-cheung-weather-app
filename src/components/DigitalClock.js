import React, { useState, useEffect } from "react";
import "../styles/digital-clock.css";

function DigitalClock() {
  const [time, setTime] = useState(new Date(Date.now()));

  useEffect(() => {
    const timeId = setInterval(() => setTime(new Date(Date.now())), 1000);
    return () => {
      clearInterval(timeId);
    };
  });

  return <div className="digital-clock"> {time.toLocaleTimeString()} </div>;
}

export default DigitalClock;
