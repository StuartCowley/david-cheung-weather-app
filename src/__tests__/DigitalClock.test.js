import React from "react";
import { render, screen } from "@testing-library/react";
import DigitalClock from "../components/DigitalClock";
import renderer from "react-test-renderer";

describe("DigitalClock", () => {
  test("Renders as expected", () => {
    Date.now = jest.fn(() => 1482363367071);

    const rendered = renderer.create(<DigitalClock />);

    expect(rendered).toMatchSnapshot();
  });
});
