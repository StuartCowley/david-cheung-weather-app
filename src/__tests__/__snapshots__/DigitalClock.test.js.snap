// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DigitalClock Renders as expected 1`] = `
<div
  className="digital-clock"
>
   
  11:36:07 PM
   
</div>
`;
